src/typescript/index.test.ts

```diff
@@ -0,0 +1,143 @@
+    expect(api.isDefinitelyTyped({ name: 'algoliasearch' })).toBe(undefined);
+
+    await api.loadTypesIndex();
+    expect(api.typesCache).toHaveProperty('algoliasearch');
+    expect(api.typesCache).not.toHaveProperty('algoliasearch/lite');
+
+    expect(api.typesCache.algoliasearch).toBe('algoliasearch');
+    expect(api.typesCache['algoliasearch/lite']).toBe(undefined);
+    expect(api.typesCache.doesnotexist).toBe(undefined);
+
+    expect(api.isDefinitelyTyped({ name: 'algoliasearch' })).toBe(
+      'algoliasearch'
+    );
+  });
+});
+
+describe('getTypeScriptSupport()', () => {
+  it('If types are already calculated - return early', async () => {
+    const typesSupport = await api.getTypeScriptSupport({
+      name: 'Has Types',
+      types: { ts: 'included' },
+      version: '1.0',
+    });
+
+    expect(typesSupport).toEqual({ types: { ts: 'included' } });
+  });
+
+  it('Handles not having any possible TS types', async () => {
+    const typesSupport = await api.getTypeScriptSupport({
+      name: 'my-lib',
+      types: { ts: false },
+      version: '1.0',
+    });
+    expect(typesSupport).toEqual({ types: { ts: false } });
+  });
+
+  describe('Definitely Typed', () => {
+    it('Checks for @types/[name]', async () => {
+      const atTypesSupport = await api.getTypeScriptSupport({
+        name: 'lodash.valuesin',
+        types: { ts: false },
+        version: '1.0',
+      });
+      expect(atTypesSupport).toEqual({
+        types: {
+          ts: 'definitely-typed',
+          definitelyTyped: '@types/lodash.valuesin',
+        },
+      });
+    });
+
+    it('Checks for @types/[scope__name]', async () => {
+      const atTypesSupport = await api.getTypeScriptSupport({
+        name: '@mapbox/geojson-area',
+        types: { ts: false },
+        version: '1.0',
+      });
+      expect(atTypesSupport).toEqual({
+        types: {
+          ts: 'definitely-typed',
+          definitelyTyped: '@types/mapbox__geojson-area',
+        },
+      });
+
+      const atTypesSupport2 = await api.getTypeScriptSupport({
+        name: '@reach/router',
+        types: { ts: false },
+        version: '1.0',
+      });
+      expect(atTypesSupport2).toEqual({
+        types: {
+          ts: 'definitely-typed',
+          definitelyTyped: '@types/reach__router',
+        },
+      });
+    });
+  });
+
+  describe('unpkg', () => {
+    it('Checks for a d.ts resolved version of main', async () => {
+      // @ts-expect-error
+      npm.validatePackageExists.mockResolvedValue(false);
+      // @ts-expect-error
+      fileExistsInUnpkg.mockResolvedValue(true);
+
+      const typesSupport = await api.getTypeScriptSupport({
+        name: 'my-lib',
+        types: { ts: { possible: true, dtsMain: 'main.d.ts' } },
+        version: '1.0.0',
+      });
+      expect(typesSupport).toEqual({ types: { ts: 'included' } });
+    });
+  });
+
+  // TO DO : reup this
+  // adescribe('FilesList', () => {
```

Comment:
what's missing to add these tests back?

Reply:
In the aforementioned PR I was prefetching the list of files beforehand but it requires more work and it's not the slowest part of the code